### ISSUE TITLE
Redis instrumentation payload for lua scripts.

### ISSUE THREAD

**Issue by alanhala** [appsignal/appsignal-ruby#726]
In our application we have some Lua scripts for Redis and I found that they are weirdly displayed in the Slow Events page. Here's an example of one:
```
eval ? ? ? ? ? ? ? ?
```
The reason is because the redis client is called with eval, sending the script as first argument and then the remaining arguments would be the lua script ones. Unfortunately the only way to fix this is by overriding the `RedisIntegration`. Is there any way to skip monkey patching?

**Comment by tombruijn** [comment]
Hi @alanhala, I think what's happening here, is that we consider the entire lua script given to the eval command as arguments. We then filter out those arguments later, as we do with any other arguments for commands.

What behavior would you expect here?
Is the first argument always the script, e.g.: `> eval "return {KEYS[1],KEYS[2],ARGV[1],ARGV[2]}" 2 key1 key2 first second`, and do you want to see that script in the event details?

**Comment by alanhala** [comment]
> Hi @alanhala, I think what's happening here, is that we consider the entire lua script given to the eval command as arguments. We then filter out those arguments later, as we do with any other arguments for commands.
> 
> What behavior would you expect here?
> Is the first argument always the script, e.g.: `> eval "return {KEYS[1],KEYS[2],ARGV[1],ARGV[2]}" 2 key1 key2 first second`, and do you want to see that script in the event details?

Yeah, correct. The first argument is always the script and then the remaining ones are the arguments passed to the script.

To be honest, I'm not sure. The workaround that I did was basically taking the string and mapping it to a symbol that represents the script name and we added a fallback to sending the whole script, just in case.

But now that I'm thinking while writing maybe a better solution would be to handle this the same way the `sql.active_record` is handled. The event title would be the id and the body the query performed without arguments.
```ruby
def process(commands, &block)
  sanitized_commands = commands.map do |command, *args|
    if command == :eval
      args[0]
    else
      command
    end
  end.join("\n")

  Appsignal.instrument "query.redis", id, sanitized_commands do
    super
  end
end
```

**Comment by alanhala** [comment]
Another thing that I realized is that you hook into the `#process` method, which in some cases can receive more than one commands (only when you call the redis client using the pipelined stuff), making the previous snippet to behave weirdly as scripts might contain "\n" too. So I was wondering if it would be better to hook into the `write` method which actually receives only one command to be executed.

What do you think about that?

**Comment by tombruijn** [comment]
The fix has been released in Ruby gem 3.0.6. Thanks @alanhala !
